src/resources/Tag.js

```diff
@@ -40,8 +43,33 @@ class Tag extends Resource {
   fetch() {
     return this.client.get(this.href)
       .then(response => response.json())
-      .then(tag => new Tag(this.client, this, tag));
+      .then(tag => new Tag(this.client, { ...this, ...tag }));
+  }
+
+  /**
+   * Saves data for a tag via the client
+   * @returns {Promise} if successful returns a tag with the id included
+   */
+  create() {
+    const { client, hydrated, customerCode, ...body } = this;
+    return this.client.post(`/1/${this.customerCode}/tags/`, { body })
+      .then(response => response.headers.get('location'))
+      .then((href) => {
+        const match = /\/\d+\/\S+\/tags\/(\d+)/.exec(href);
```

Comment:
Should we just alter the response from the API to include an `Id` header, then get the ID from that instead of parsing it out of `Location`?